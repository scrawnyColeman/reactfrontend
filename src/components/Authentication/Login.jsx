import React, { useState } from 'react';
import styled from 'styled-components';
import LoginForm from './LoginForm';
import RegisterForm from './RegisterForm';
import Button from '../Button/component';

const StyledWrapper = styled.div`
    display: grid;
    grid-template-rows: max-content max-content;
    grid-gap: 30px;
    margin: 20vh auto;
`;
const StyledLogoContainer = styled.div`
    margin: 0 auto;
`;
const StyledFormContainer = styled.div`
    margin: auto;
`;
const StyledName = styled.div`
    display: block;
    font-size: 3rem;
`;

const Login = () => {
    const [hasAccount, setHasAccount] = useState(false);

    return (
        <StyledWrapper>
            <StyledLogoContainer>
                <StyledName>NJOY LEARNING</StyledName>
            </StyledLogoContainer>
            <StyledFormContainer>
                {hasAccount ? <LoginForm /> : <RegisterForm />}
                {hasAccount && (
                    <span>
                        Don't have an account?
                        <Button onClick={() => setHasAccount(false)} text="Register" size="large" variant="text" />
                    </span>
                )}
                {!hasAccount && (
                    <span>
                        Already have an account?
                        <Button onClick={() => setHasAccount(true)} text="Log in" size="small" variant="text" />
                    </span>
                )}
            </StyledFormContainer>
        </StyledWrapper>
    );
};
export default Login;
