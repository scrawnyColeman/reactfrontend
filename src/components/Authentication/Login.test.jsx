import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-styled-components';
import React from 'react';
import Login from './Login';

Enzyme.configure({ adapter: new Adapter() });

describe('Login component', () => {
    test('component matches snapshot', () => {
        const wrapper = shallow(<Login />);
        expect(wrapper).toMatchSnapshot();
    });
});
