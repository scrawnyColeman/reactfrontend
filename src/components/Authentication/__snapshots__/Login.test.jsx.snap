// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component component matches snapshot 1`] = `
<styled.div>
  <LoginError
    message="Account information is invalid."
  />
  <styled.div>
    <styled.div>
      NJOY LEARNING
    </styled.div>
  </styled.div>
  <styled.div>
    <RegisterForm />
    <span>
      Already have an account?
      <Button
        onClick={[Function]}
        size="small"
        text="Log in"
        variant="text"
      />
    </span>
  </styled.div>
</styled.div>
`;
